command/src/com/mirth/connect/cli/CommandLineInterface.java

```diff
@@ -179,10 +180,21 @@ private void runShell(String server, String user, String password, String script
             client = new Client(server);
             this.debug = debug;
 
-            LoginStatus loginStatus = client.login(user, password);
-
-            if (loginStatus.getStatus() != LoginStatus.Status.SUCCESS) {
-                error("Could not login to server.", null);
+            LoginStatus loginStatus;
+            try {
+                loginStatus = client.login(user, password);
+            } catch (UnauthorizedException ex) {
+                if (ex.getResponse() instanceof LoginStatus status) { 
+                    loginStatus = status; 
+                } 
+                else { 
+                    error("Could not login to server.", ex);
+                    return;
+                }
+            }
+            
+            if (!loginStatus.isSuccess()) {
+                error("Could not login to server. Status: " + loginStatus.getStatus(), null);
                 return;

```

Comment:
```suggestion
                error("Could not login to server. Please check your username and password and try again.", null);
                System.exit(77);
```

77 is the unix exit code for EX_NOPERM. This code is already using exit code 2 for usage errors.

Reply:
@tonygermano should we follow the same exit code structure when returning from the runScript/runConsole? 

Exceptions on those functions are currently being ignored and the command just continues.